internal/show/versions.go

```diff
@@ -67,19 +94,119 @@ func printDeploymentVersions(clusterInfo *cluster.Info, printer *table.Printer,
 		}
 
 		version, repository := images.ParseOperatorImage(deployment.Spec.Template.Spec.Containers[0].Image)
-		printer.Add(component, repository, version)
+
+		runningVersion, err := getVersionForComponent(clusterInfo, component,
+			labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels).String())
+		if err != nil {
+			return err
+		}
+
+		printer.Add(component, repository, version, runningVersion)
 	}
 
 	return nil
 }
 
+func getVersionForComponent(clusterInfo *cluster.Info, component, labelSelector string) (string, error) {
+	podsClient := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Pods(constants.OperatorNamespace)
+	podList, err := podsClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+
+	if err != nil || len(podList.Items) < 1 {
+		return "", errors.Wrapf(err, "Failed to find pods for component %s", component)
+	}
+
+	// Try all pods
+	for i := range podList.Items {
+		pod := podList.Items[i]
+		podVersion, _ := getVersionFromPodBinary(&pod, clusterInfo, component)
+
+		if podVersion != "" {
+			return podVersion, nil
+		}
```

Comment:
We shouldn't ignore the return error.

```suggestion
		podVersion, err := getVersionFromPodBinary(&pod, clusterInfo, component)

		if err != nil || podVersion != "" {
			return podVersion, err
		}
```

Reply:
We want to ignore error coz we will try getting version from log if failed to fetch from binary.

Reply:
But an error here should indicate a non-recoverable failure, eg K8s API server not available, whereas if the error hadn't occurred then we would've determined the version. If no version is found then try the pod log. The latter case should only occur for a pre 0.16 version.